sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java

```diff
@@ -8,43 +8,110 @@
 import com.azure.security.attestation.implementation.AttestationsImpl;
+import com.azure.security.attestation.implementation.MetadataConfigurationsImpl;
+import com.azure.security.attestation.implementation.SigningCertificatesImpl;
 import com.azure.security.attestation.models.AttestOpenEnclaveRequest;
 import com.azure.security.attestation.models.AttestSgxEnclaveRequest;
 import com.azure.security.attestation.models.AttestationResponse;
+import com.azure.security.attestation.models.AttestationSigner;
 import com.azure.security.attestation.models.CloudErrorException;
 import reactor.core.publisher.Mono;
 
 import java.nio.charset.StandardCharsets;
 import java.util.Objects;
 
+import static com.azure.core.util.FluxUtil.withContext;
+
 /** Initializes a new instance of the asynchronous AzureAttestationRestClient type. */
-@ServiceClient(builder = AttestationClientBuilder.class, isAsync = true)
+@ServiceClient(builder = AttestationAsyncClientBuilder.class, isAsync = true)
 public final class AttestationAsyncClient {
-    private final AttestationsImpl serviceClient;
+    private final AttestationsImpl attestImpl;
+    private final MetadataConfigurationsImpl metadataImpl;
+    private final SigningCertificatesImpl signerImpl;
 
     /**
      * Initializes an instance of Attestations client.
      *
-     * @param serviceClient the service client implementation.
+     * @param clientImpl the service client implementation.
+     */
+    AttestationAsyncClient(AttestationClientImpl clientImpl) {
+        this.attestImpl = clientImpl.getAttestations();
+        this.metadataImpl = clientImpl.getMetadataConfigurations();
+        this.signerImpl = clientImpl.getSigningCertificates();
+    }
+
+    /**
+     * Retrieves metadata about the attestation signing keys in use by the attestation service.
+     *
+     * @param context - Context for the operation
+     * @throws CloudErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return any object.
+     */
+//    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Object>> getOpenIdMetadataWithResponse(Context context) {
+        return this.metadataImpl.getWithResponseAsync(context);
+    }
+
+    /**
+     * Retrieves metadata about the attestation signing keys in use by the attestation service.
+     *
+     * @throws CloudErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return any object.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Object> getOpenIdMetadata() {
+        // Forward the getOpenIdMetadata to the getOpenIdMetadataWithResponse API implementation.
+        return withContext(context -> this.getOpenIdMetadataWithResponse(context))
```

Comment:
> return withContext(context -> this.getOpenIdMetadataWithResponse(context)) [](http://example.com/codeflow?start=7&length=75)

You will have to add a service method for the WithResponse overload as well (w/o context) just like the one above.

Reply:
Ah, missed that (mostly because it wasn't in the generated code).